### DISCUSSION TITLE
Is there any chance I can config entrypoints elsewhere?

### DISCUSSION THREAD

**Discussion by MiloWang2048** [privatenumber/pkgroll#58] [Q&A] [answered]
### Question

I'm working on a monorepo project and I need to point one of my package A onto it's source code like
```json
  "exports": {
    ".": {
      "default": "./src/index.ts"
    }
  },
```
so when deving other packages I don't need to open a watch&compile process on A; they get realtime type and reference just out of box.

BUT when it goes into production eventually I need to build this thing to js, and with pkgroll's config detection there will be no build at all. If I can configure it elsewhere, problem solved. So, is there such a way to do this?

BTW pkgroll is really brilliant. I love it really.

**Comment by privatenumber** [comment, marked as answer]
If I understand your use-case, you want it to use `.ts` files in development, and `.js` in production? If so, that's exactly what export maps are for.

Checkout this example where I use import maps (same thing as export maps but for internal paths) to specify a development path under the `development` condition: https://github.com/privatenumber/get-tsconfig/blob/d1c46c04f200a8239f3e5734c42011de1ea11380/package.json#L39

To access it, I just need to pass in `development` to node: `node -C development ./entry.js`

**Reply by MiloWang2048** [reply]
Well, rules `"imports"` field specified can only be applied inside of the pkg A, if I recall it; but the problem did solved by conditional exporting: I need type imports only in development and don't need dts files at runtime, so:
```json
  "exports": {
    ".": {
      "types": "./src/index.ts", // development
      "default": "./dist/index.js" // build
    }
  },
```
works just fine for me. Thanks a lot!
